### DISCUSSION TITLE
Can we use noncubic domain with periodicity?

### DISCUSSION THREAD

**Discussion by skiyohara** [CGAL/cgal#8145] [Q&A] [answered]
Dear CGAL developers and users,

I'm a new to cgal and want to use non-cubic domain with periodicity.
In the latest version (5.6.1), cgal implicitly defines a domain using three vectors in 3D , **a**=(a,0,0), **b**=(0,b,0), and **c**=(0,0,c), where a, b, c are the lengths of the cubic.
In my case, i want to define a domain with **a**=(a1,a2,a3), **b**=(b1,b2,b3), and **c**=(c1,c2,c3).
Can I use cgal under the setting? or do I need to rewrite the cgal code myself?

Any help would be greatly appreciated.

**Comment by MaelRL** [comment, marked as answer]
Currently there is no way to use domains that are not axis-aligned boxes.

There is a WIP branch based on https://drops.dagstuhl.de/entities/document/10.4230/LIPIcs.ESA.2020.75 that would allow your kind of domain, but it's not in a usable state as of now.
